pkg/providers/provider.go

```diff
@@ -32,6 +32,7 @@ type Provider interface {
 	MachineConfigs() []MachineConfig
 	ValidateNewSpec(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 	GenerateMHC() ([]byte, error)
+	ApplyWorkerNodeGroupTaints(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster) error
```

Comment:
It's unfortunate we need to this at the provider level
One more reason to try to abstract it more in the api (see me other comment about this)